pkg/tempo/config.go

```diff
@@ -50,23 +85,9 @@ type PushConfig struct {
 	RetryOnFailure     map[string]interface{} `yaml:"retry_on_failure,omitempty"` // https://github.com/open-telemetry/opentelemetry-collector/blob/1962d7cd2b371129394b0242b120835e44840192/exporter/exporterhelper/queued_retry.go#L54
 }
 
-// UnmarshalYAML implements yaml.Unmarshaler.
-func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	// If the Config is unmarshaled, it's present in the config and should be
-	// enabled.
-	c.Enabled = true
-
-	type plain Config
-	return unmarshal((*plain)(c))
-}
-
-func (c *Config) otelConfig() (*configmodels.Config, error) {
+func (c *InstanceConfig) otelConfig() (*configmodels.Config, error) {
 	otelMapStructure := map[string]interface{}{}
 
-	if !c.Enabled {
```

Comment:
This code path should never have hit anyway, cmd/agent doesn't instantiate the tempo subsystem unless it's enabled.